File: cpp/49-GroupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> umap;
        for (string str : strs) {
            string tmp = str;
            sort(tmp.begin(), tmp.end());
            umap[tmp].push_back(str);
        }
        for (auto pair : umap) {
            res.push_back(pair.second);
        }
        return res;
    }
};

File: Java/49-GroupAnagrams.java
public class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs == null || strs.length == 0)   
            return new ArrayList<List<String>>();
        Map<String, List<String>> map = new HashMap<>();
        for (String string : strs){
            char[] ca = string.toCharArray();
            Arrays.sort(ca);
            String keyString = String.valueOf(ca);
            if (!map.containsKey(keyString)){
                map.put(keyString, new ArrayList<String>());
            }
            map.get(keyString).add(string);
        }
        return new ArrayList<List<String>>(map.values());
    }
}